dask_planner/src/sql.rs

```diff
@@ -488,11 +512,12 @@ impl DaskSQLContext {
     ) -> PyResult<logical::PyLogicalPlan> {
         // Certain queries cannot be optimized. Ex: `EXPLAIN SELECT * FROM test` simply return those plans as is
         let mut visitor = OptimizablePlanVisitor {};
+        let statistics = self.get_table_statistics();
 
         match existing_plan.original_plan.accept(&mut visitor) {
             Ok(valid) => {
                 if valid {
-                    optimizer::DaskSqlOptimizer::new(true)
+                    optimizer::DaskSqlOptimizer::new(true, statistics)
```

Comment:
I'm a bit conflicted b/w pre-loading every single table stat in a dictionary and passing the whole thing over to the optimizer config vs getting it from the specific rule that actually needs it.

Reply:
Elaborating a bit more, being able to do something like this: https://github.com/dask-contrib/dask-sql/pull/1027/files#diff-9b258b399cd311b18a5b265d5bbdd0a73eb211a7f0975e967a4b17eae9a804a6R400 by just making modifications to `DaskTableSource`. Though I'm not familiar enough to comment on whether that approach will work purely in dask-sql without depending on any additional improvements to datafusion.

Reply:
Okay, thanks! I've updated the code with modifications to `DaskTableSource`. In PR #1027, it looks like it would work by having a function:
```
fn get_table_size(plan: &LogicalPlan) -> Option<usize> {
    match plan {
        LogicalPlan::TableScan(scan) => {
            let source = scan
                .source
                .as_any()
                .downcast_ref::<DaskTableSource>()
                .expect("should be a DaskTableSource");
            if let Some(stats) = source.statistics() {
                Some(stats.get_row_count() as usize)
            } else {
                None
            }
        }
        _ => get_table_size(plan.inputs()[0]),
    }
}
```
in the `join_reorder.rs` file. The function appears to work as desired, at least within the Join Reordering PR 🙂

Let me know what you think.

Reply:
looks good to me!